### PR TITLE
Block Support: Add border radius support

### PR DESCRIPTION
**Pull request by aaronrobertshaw**
#### Background

To allow designers to produce different kinds of patterns using different blocks there is a need to provide control over a block's border radius. Some example use cases are:

* https://github.com/WordPress/gutenberg/issues/22071
* https://github.com/WordPress/gutenberg/issues/26559
* https://github.com/WordPress/gutenberg/issues/26556

Initially, this was attempted via an adhoc approach adding the feature directly to the search block https://github.com/WordPress/gutenberg/pull/25553. This PR adds border radius to the block support tools so that it may be leveraged by other blocks. 

The intention is to also expand this to include border width and style options. There is also the possibility to allow selection of which borders show i.e. all, top, right, bottom, left, none.

#### Changes

* Adds new block support for borders similar to the typography and color tools
* Adds border radius hook to the borders support providing; inspector controls, attributes and styles

## How has this been tested?
Tests: 
* `packages/block-editor/src/hooks/tests/style.js`

Manually, by opting in to border radius support for test blocks such as the group and image blocks. 

#### Testing Setup

To be able to manually test this you will need a block that has opted into the border radius support. The group block is probably easiest to do this with and will be used in the testing instructions that follow.

To opt into border radius support for the group block simply add into the `supports` setting in
`packages/block-library/src/group/block.json` the following code:

```
        "__experimentalBorder": {
            "radius": true
        }
```

#### Testing Instructions
1. Create a new post and add a group block, setting a prominent background color on the group
2. Selecting the group block, in the inspector panel, drag the border radius slider and confirm the block reflects this
3. Type a border radius value into the text input and confirm that is reflected
4. Click the reset button and there should no longer be a border radius applied to the block
5. Set a border radius value again and then save the post
6. Confirm border radius values are present on the frontend

## Screenshots 
BorderRadiusSupport

## Types of changes
New feature.

## Next Step
* Update `docs/designers-developers/developers/block-api/block-supports.md` once stable.

## Future Iterations:
* Add support for border width
* Add support for border style
* Add support for top, right, bottom, left borders

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by simison** [comment]
This will be fantastic for gallery block 🎉

**Comment by glendaviesnz** [comment]
This tests well for me. Although it does make for a large PR by including the Search block changes I think this makes it easier for testing to have an implementation included.

**Comment by aaronrobertshaw** [comment]
Agreed. The PR did grow larger than expected. I can split this into two to make it easier to review if you'd like.

**Comment by glendaviesnz** [comment]
> Agreed. The PR did grow larger than expected. I can split this into two to make it easier to review if you'd like.

I find it easier to review with the search block implementation included.

**Comment by aaronrobertshaw** [comment]
There are essentially two areas of concern in this PR. Adding block support for border radius styles and leveraging this within the search block. A quick summary of the changes to each file can be found below. Hopefully that provides a little more context for the changes.

## Adding Border Related Block Support

##### `lib/block-supports/border.php`

Provides functions that will be called in `lib/block-supports/index.php` to register and apply a new set of block support tools related to borders. This is intended to be similar to typography. 

The register function handles injecting any new attributes required. The apply function handles generating CSS classes or inline styles.

##### `lib/block-support/index.php`

Call the new toolset’s functions to register and apply border block support.

##### `lib/experimental-default-theme.json`

Enables support for border styles with the default theme.json.

##### `lib/global-styles.php`

Add border styles and radius to the global styles config.

##### `lib/load.php`

Loads new border block support.

##### `packages/block-editor/src/hooks/border-radius.js`

This is the main file that provides the border radius block support. It defines functions and adds filters to inject attributes and tweak props while in the block editor. Also defines a helper for if the feature is enabled and the edit component for controlling the new border radius setting.

##### `packages/block-editor/src/hooks/border.js`

This handles the new set of border related block support tools. It defines a panel for the InspectorControls and helpers for whether the tools are enabled or are supported.  

##### `packages/block-editor/src/hooks/style.js`

The changes here handle adding the new border block support controls to the editor for blocks when appropriate.

##### `packages/block-editor/src/hooks/test/border-radius.js`

Includes tests to check injection of attributes and manipulation of props.

##### `packages/block-editor/src/hooks/test/style.js`

Updates test to ensure the border radius inline style is generated correctly.

##### `packages/blocks/src/api/constants.js`

Adds border radius to the experimental style properties API constant.

## Search Block Border Radius 

##### `packages/block-library/src/search/block.json`

Opts in to using the experimental border radius feature.

##### `packages/block-library/src/search/edit.js`

These changes centre around using the new border radius attribute to provide border radius to not only the main wrapper but to the inner components as needed. This involves calculating adjusted inner border radius values to avoid “fat corners” and maintain visual consistency.

##### `packages/block-library/src/search/icons.js`

Fixes errors in console related to unknown prop `stroke-width`. 

##### `packages/block-library/src/search/index.php`

Updates to the frontend markup generation to apply an adjusted border radius to the inner components as required depending on button positioning for the search block.

File: lib/experimental-default-theme.json
Comment (nosolosw) [inline lib/experimental-default-theme.json:169]: How about using `border` instead, as we do in `lib/block-supports/border.php` and follows the naming schema of the other groups (`typography`, `color`, `spacing`)?

The properties within could also remove the `border` infix and be something like `customRadius`.

File: packages/block-library/src/search/block.json
Comment (nosolosw) [inline packages/block-library/src/search/block.json:38]: I've noticed the other support flags have been stabilized and no longer have the `__experimental` prefix, so I assume we are happy with the direction and new additions as this can also land as stable cc @jorgefilipecosta @youknowriad @ItsJonQ for thoughts.

Comment (nosolosw) [reply packages/block-library/src/search/block.json:38]: Also, if we're to expand this with other border properties: does it make sense that the API here allows for grouping things like we do in color?

```
__experimentalBorder: {
    radius: ..., 
    width: ...,
    style: ...
}
```

File: lib/experimental-default-theme.json
Comment (aaronrobertshaw) [reply lib/experimental-default-theme.json]: That makes more sense. Done ✅

File: packages/block-library/src/search/block.json
Comment (aaronrobertshaw) [reply packages/block-library/src/search/block.json]: I've updated this to allow for a collection of border related features as per your suggestion. The `__experimental` prefix has been left for the time being.

**Comment by stokesman** [comment]
Seems like something that really ought to be implemented. For one thing, this is how the border-radius on buttons should be made available since some folks definitely want to be able to disable that.

Also, here's a few feature requests that would be facilitated by the block supports work in this PR: #26556 #26579.

**Comment by aaronrobertshaw** [comment]
Since this PR was created, block support has evolved quite a bit. I've rebased this and updated it to work with the latest approach to providing block support.

It would be great if someone can give this another review.

File: packages/block-library/src/search/index.php
Comment (youknowriad) [inline packages/block-library/src/search/index.php:169]: I was hoping we'd avoid the need for any particular code on blocks when you enable "support" for this? why is this needed?

File: packages/block-library/src/search/edit.js
Comment (youknowriad) [inline packages/block-library/src/search/edit.js:62]: Same here, it feels there are a lot of changes here where I'd have expected everything to be absorbed by the support flag and hooks?

File: packages/block-library/src/search/index.php
Comment (aaronrobertshaw) [reply packages/block-library/src/search/index.php:169]: This was required to maintain visual consistency between the border radius for the block itself and its inner elements when appropriate. 

When the search block is configured to position the button "inside", the border radius that the search block opted into needs to be applied not only to the wrapper around the search input and button but also to the input and button inner elements. This is complicated a little further as they need to be adjusted for the inner padding of the wrapper so that the border radius can match visually with the wrapper's border radius and we do not get "fat" corners.

This addresses an [issue highlighted](https://github.com/WordPress/gutenberg/pull/25203#issuecomment-689870244) with the original PR that sparked this one.

Comment (youknowriad) [reply packages/block-library/src/search/index.php:169]: Do you think it's possible to leverage some CSS magic to fix that instead? I'm asking cause this makes me wonder whether the block support is the right approach for this particular block or not. Basically, why is it even a block support if it requires such custom code, it removes the declarativeness of the block support. 

Block supports are and should be deterministic IMO: you apply them, you get a block radius automatically applied without custom tweaks. So it makes wonder whether block radius support flag is better suited for simpler blocks say button for instance or a group...

Comment (aaronrobertshaw) [reply packages/block-library/src/search/index.php:169]: > Do you think it's possible to leverage some CSS magic to fix that instead?

That's a good question, I'll look into that. Not being able to use an inherited value within `calc()` might make it tricky. Maybe CSS variables could offer something. 

One other thing that could complicate this, arose from some internal discussions, which raised the possibility of adding spacing/padding and border width support, both of which would affect the calculations for the inner border radii.

> I'm asking cause this makes me wonder whether the block support is the right approach for this particular block or not. Basically, why is it even a block support if it requires such custom code, it removes the declarativeness of the block support.

That's true regarding the search block itself. However, as you mention, simpler blocks could benefit from the border radius support and keep that declarative nature. If the search block still needs to offer a border radius feature, would it not be better to reduce duplication of code and leverage the block support for border radius?

We still have the initial PRs trialling adding the border radius feature for the search block in an ad-hoc manner, https://github.com/WordPress/gutenberg/pull/25553 & https://github.com/WordPress/gutenberg/pull/25203.

If you think that approach is actually the better option for the search block, I'm happy to remove the modifications to that block from this PR. It would be a shame though not to proceed with adding border radius support that other blocks can start to use.

Comment (youknowriad) [reply packages/block-library/src/search/index.php:169]: I'd love some thoughts from friends @nosolosw @jorgefilipecosta @mcsf @mtias

Comment (aaronrobertshaw) [reply packages/block-library/src/search/index.php:169]: I’ve tried utilising a CSS approach to achieve the desired results in the search block however it leaves a lot to be desired. 

**Approach:**
* Each block can have it’s own border radius. This means each block needs it’s own CSS rule to set a  `--border-radius` CSS variable within a selector that targets that block only.
* With a properly scoped CSS variable, the input and button inner components could have generic CSS rules that would inherit that value using something like: 
`border-radius: max( calc( var( --border-radius ) - 4px ), 1 );`
* The downside is that each block would now need to generate its own `<style>`  and CSS rule to achieve the above. Still moving away from the ideal where use of the border radius support for the search block was only declarative.

**Alternate approaches:**
* If we used preset values in the theme.json, global styles would handle injection of those CSS variables into the stylesheet. We then lose the ability to choose custom values and adding 50 preset values isn’t an option. 
* To use the border radius block support directly on the inner components, they would need to be converted to blocks that could opt in to the support. Operating independently, it would be up to the user to set the border radii on them, making sure they were in sync and manually calculating or visually adjusting the inner components’ radii until they achieved visual consistency with the outer wrapper radius when the button is placed inside. Not a great experience for the user.

**Summary:**

Presently, I don’t see an option to be able to provide the border radius feature on the search block as needed, without changes to the block itself. The current approach in this PR keeps those changes relatively straightforward, to a minimum and avoids duplicate code adding border radius controls etc that would increase the maintenance burden.

Happy to hear others’ opinions and any ideas to address concerns. For the moment though my vote would be to proceed with the current approach.

**Comment by paaljoachim** [comment]
Hey Aaron

Nice job!

This is what I see:
Screen Shot 2020-11-07 at 00 58 07

The sidebar settings do not contain Border Settings.

There are two toggles in the toolbar, and one drop down. The user will need to click the icons in the toolbar to try them out. 

For an easier overview the toggles can be added to the sidebar panel. Containing easy to see labels.

Here is one suggestion:
Search-block-options

We have then only the one unique Search block icon in the toolbar. Keeping a clear difference of location between the toggles and the drop down option.

**Comment by jasmussen** [comment]
I'll share some more thoughts next week, I think there's a chance to systematize this, as I'd agree generally that the addition feels a little random, even if useful. We need to think in terms of how this can be applied to all blocks, and form a coherent system.

> For an easier overview the toggles can be added to the sidebar panel. Containing easy to see labels.

I would not do this, though. [The block itself is the primary interface](https://developer.wordpress.org/block-editor/designers/block-design/), and anything you put in the sidebar should be considered secondary. While there's an argument to be made that some of those configurations could be secondary. So I'd keep the items in the toolbar as they are now, even if those options could use some systematization and cleanup separately.

**Comment by aaronrobertshaw** [comment]
Hi @paaljoachim, thanks for taking the time to review this.

> In the version I see. The sidebar settings do not contain Border Settings.

Checking out this PR branch I see the border settings in the sidebar. I've also re-built Gutenberg but cannot replicate the behaviour you describe. Below is a new gif of what I see. 

BorderRadiusSupport

In both my tests this morning, I've fired up the Gutenberg environment with this PR branch checked out already. It could be an issue with switching from master to this branch with Gutenberg env already up and being watched. I'll rebase this branch and test again.

**Comment by aaronrobertshaw** [comment]
After rebasing this, the border settings are still showing correctly for me. Switching between master and this branch did require a hard refresh of the editor though.

**Comment by jasmussen** [comment]
Some further thoughts. Most importantly, I appreciate the work on adding additional design tools to our blocks, enabling better patterns and creations. Please do keep this appreciation for the work in mind as you read the following, because the feedback is not just about this PR, but about the approach in general.

However as this accelerates, it's increasingly clear that the sidebar is not able yet to scale to accommodate these new controls:

- Controls are added seemingly randomly to random blocks
- Additional collapsible panels are added when an existing one doesn't seem appropriate
- In being given a whole panel with just one control inside, those controls are given prominence that isn't deserved

More importantly, existing patterns even if arbitrary aren't followed, so for exampe in this case, "Border settings" is an existing panel for the Buttons block, where it's the 3rd panel, but here it's the first panel. Additionally, the label case is wrong, "Border Settings" where it should be "Border settings" per past decisions. 

I've been working with some others on some systems for how controls can be added to blocks in a predictable and consistent way, it's hard work and it's taken a while and I expect and hope that it's shareable very soon. But in the mean time, we need to find a more holistic approach that is less prone to fragment the experience. That approach could start with a conversation around how the new design tool fits into global styles, and from there, how it is then surfaced on a per-block basis.

**Comment by aaronrobertshaw** [comment]
@jasmussen Thank you for sharing your thoughts and tackling these issues, it is appreciated. 👍 

Working through a solid process from initial conversations around the new design tool through to how it is surfaced, sounds great. If I can help at all with that, please let me know.

In the meantime, I can correct the panel title to use sentence case. The ordering of the panel to match other blocks might be a little trickier as it currently gets injected automatically via block support and different blocks will have a different number of panels. I'll see what I can do there to improve that situation. It would be good to still be able to finesse this to a point it could be merged.

As we build additional block patterns and discover the need for new design tooling, I'll make sure we loop in anyone we need to, into the discussion. Who would be best to include in these discussions?

**Comment by aaronrobertshaw** [comment]
> Additional collapsible panels are added when an existing one doesn't seem appropriate
> In being given a whole panel with just one control inside, those controls are given prominence that isn't deserved

For a little background here, the border settings panel was added to contain more than just the border radius in the future. The idea being we might allow for border width, style and so on as well. The block support feature was also structured with this future in mind.

> In the meantime, I can correct the panel title to use sentence case. The ordering of the panel to match other blocks might be a little trickier

I've updated this to use sentence case for the border radius panel title, its field label and the "Display settings" panel for consistency.

Unfortunately, I do not see an easy solution at the moment for standardising the order or position of the panels in the inspector controls. 

Besides different blocks requiring display of differing settings panels, the addition of these panels is segmented. Presently, any panels being injected via hooks appear before those added ad-hoc per block. Block styles are slightly different but they will be added before any other panels, if the block defines them. 

I don't think this should block this PR but we can certainly put a hold on future additions until we have a resolution on how to proceed in a consistent, scalable manner.

**Comment by apeatling** [comment]
> I would not do this, though. The block itself is the primary interface, and anything you put in the sidebar should be considered secondary. While there's an argument to be made that some of those configurations could be secondary. So I'd keep the items in the toolbar as they are now, even if those options could use some systematization and cleanup separately.

+1 on considering the sidebar as secondary. We've seen in almost all user tests on WordPress.om that users have a hard time finding anything for a block in the sidebar. They go scanning the block toolbar looking for anything related to block settings. Making the block toolbar the primary interface and only falling back to the sidebar for lesser used or power settings makes the most sense to me.

File: packages/block-library/src/search/index.php
Comment (jorgefilipecosta) [reply packages/block-library/src/search/index.php:169]: This is a very tricky case.


When we were discussing if we should apply styles directly with global styles/ block support styles or applying variables `--wp-border-radius`, that the blocks would be responsible for dealing with, for this case it seems it would be easier if we followed the variables path. But if we used variables we would have the inheritance problem e.g: applying a border-radius for a group block would be inherited by the search block inside.


Under normal circumstances when we apply a border-radius there is no expectation that the child nodes will have a border-radius and that this border-radius is calculated based on the parent value. So what we have here is not a normal border-radius.


The block.json specifies support for border-radius, when global styles adds border-radius, it will be possible to control border-radius for the search block from global styles.
The solution works very well when we apply things at the local level, but it will not work when we apply a border via global styles. The child nodes would not have the border-radius style. We would have a border-radius property applied via CSS but the block border-radius style the server sees during the block rendering would be empty. We would have an inconsistency a search block with 30px border-radius applied at block level would look different than search block with 30px applied specifically to search block at the global styles level.


I think a block should never use a style attribute as part of its rendering, ideally, everything that comes from the style attribute can also come globally from global styles. If the block accesses the style attribute for something it creates an inconsistency between what happens with local styles and what happens with global styles.


I see three possible paths forward:

- We say a generic border-radius mechanism does not fit the needs of the search block. The search block does not uses any supports for border-radius and implements a border-radius picker at the block level, saved outside of the style attribute. I guess we may reuse components from a generic border-radius picker and the search block implementation to avoid duplicate code. The search block does not support global styles border-radius. A global styles setting will not be available for the block. If themes add it via theme.json it would also not work. On the theme template, HTML file themes can add a local border-radius if they use a search block there.


- We say the search block does not support border-radius but supports a style specific to it that is applied using a CSS variable. Similar to the way link color works. We would have something like: `--wp--style--block--core--search--border-radius` or `--wp--style--border--search-border-radius` that contains the border-radius specifically for the search block. The value is saved as part of the style attribute and the attribute is simply applied via a CSS variable. The block CSS uses the variable value to do its magic apply it to child nodes and do some calculations.
Themes would not be able to use the normal border-radius on theme.json to control the search block radius but they would be able to control the border radius of the search block using something custom like `search: { borderRadius: "50px" }` or `{searchBorderRadius: "50px" }`. If we follow this path I guess the first version could have some things hardcoded as the link color does. But in the future, we would have an API that allows third parties to add this type of specific styles. Essentially blocks would be able to add global styles that make sense to them.

- Blocks can say a specific style that already exists should be applied to them using a CSS variable instead of the default mechanism. When that flag is present both global styles and the local styles instead of generating "border-radius: 50px" on the presence of a border-radius in the search block scope would automatically generate something like `--wp--block--style--core--search--border-radius`. This variable is block specific so there is no inheritance. This is a generic system any block can opt-in to use variables for specific style. Themes would be able to apply border-radius style to search via theme.json as they apply to any other block (provided the style is scoped to the search block).

I am not sure what is the best path forward as none of the solutions seem perfect. The safest bet is the first option, it does not expose anything as part of our API's, but it is also the least powerful and discards totally the possibility of controlling border-radius from the search block from global styles/theme.json.
Any thoughts on this @mtias, @shaunandrews, @aristath, @nosolosw, @youknowriad?

Comment (aristath) [reply packages/block-library/src/search/index.php:169]: I honestly don't see any benefit to adding a border-radius control to the search block...
Border-radius is something that should be consistent on all elements on a page. Having different border-radii in different elements on a page creates a visual discontinuity and is a big no-no design-wise. It's OK to have some elements rounded while others are square(ish), but having a border-radius of 7px on some elements while other elements on the same page have 3px is not something that should be done. 
With that in mind, I'd say that themes should add a global border-radius in their theme.json file and use it consistently across the board for all their blocks - directly in their CSS files. The search block can have a separate "rounded" style, but there's no need for a border-radius control... It can simply be a block-style.

Comment (aaronrobertshaw) [reply packages/block-library/src/search/index.php:169]: I agree with the need for consistency of border radius across different elements on the page. 

One point worth noting here though is that differing border-radius can actually be needed to maintain visual consistency. For example, so that nested border radii actually appear consistent with each other rather than having "fat corners". 

An image highlighting this can be found in [this comment](https://github.com/WordPress/gutenberg/pull/25203#issuecomment-689870244).

Comment (aristath) [reply packages/block-library/src/search/index.php:169]: Yeah, but that is for elements in the same block.... So it's something that can be done directly in the theme's stylesheet, and values can be calculated easily via CSS calc. However, I was referring to cross-block inconsistencies. We can't have galleries use a 10px border-radius, search using a 6px radius, 1 group using 20px and search using something different... There needs to be consistency. And that consistency can be accomplished with simple CSS styles on the theme itself - without the need for any extra controls.

Adding a control to change the border-radius of a specific element will ultimately lead to visual inconsistencies since it won't follow the overall tone set by the theme. The search-block should either follow the global theme's border-radius (taking into account nested border-radii where needed in the theme's styles), or have a rounded style. Anything other than that would cause visual inconsistencies. So I really don't see why this can't be a simple default/rounded block-style...

Comment (aaronrobertshaw) [reply packages/block-library/src/search/index.php:169]: >  However, I was referring to cross-block inconsistencies. 

Sorry, that is what I meant by I agree with the need for consistency across different elements on the page. I was only trying to highlight that not all border radius values would be the same everywhere.

> Adding a control to change the border-radius of a specific element will ultimately lead to visual inconsistencies since it won't follow the overall tone set by the theme.

It could well do, but to play devil's advocate, should the end user have the freedom? We allow the option for users to choose random colors potentially destroying the look and feel of a theme.

> The search-block should either follow the global theme's border-radius (taking into account nested border-radii where needed in the theme's styles), or have a rounded style. Anything other than that would cause visual inconsistencies. So I really don't see why this can't be a simple default/rounded block-style...

This arose while attempting to build new block patterns that included border radii within the design for the pattern. I'll have to confirm the pattern previews can actually inherit from the global styles. If we are essentially tying the end user's hands with regards to the border radius values once the pattern is inserted, we would need to limit any disjoint between what they see in the preview vs the end result. That should be the case anyway but would be of lesser consequence if they have full control.

Comment (aristath) [reply packages/block-library/src/search/index.php:169]: > but to play devil's advocate, should the end user have the freedom?

We're about to enter a more philosophical discussion if we go down that path, but there is such a thing as too much control and a difference between control and enabling poor design choices. 

> This arose while attempting to build new block patterns that included border radii within the design for the pattern.

Block-patterns _can_ use block-styles. And a block-style simply adds a CSS-class to the block. So having a `is-style-round` class added to the search block is all that's really needed - which can be accomplished with a plain block-style selector like buttons have, and simple CSS :wink:

**Comment by aaronrobertshaw** [comment]
@aristath I've created a PR (https://github.com/WordPress/gutenberg/pull/27220) to add a "rounded" block style for the search block. Does this fit with what you envisioned?

If the block style approach is acceptable, I can remove the search block related changes from this PR. The question then is should we still add border radius block support? Others [have expressed some interest](https://github.com/WordPress/gutenberg/pull/25791#issuecomment-704093356) in this and it does provide an initial base to which border width support etc can also be added.

Ultimately, do we proceed with this PR or abandon it? 

cc @apeatling @glendaviesnz

**Comment by paaljoachim** [comment]
Border radius control feels more like a standard control that can be added to various blocks where the user can choose how much by slider or number to curve the corners of a block. For instance in the Group, Columns, Heading or Search blocks.

I feel that styles is more of a unique layout option not easily accomplished by separate controls. 
One could of course adjust the Social Links icon block and remove the pill shape and instead add in a border radius control. 

Here is an example from the Buttons block using Twenty Twenty theme:

Screen Shot 2020-11-24 at 11 29 22

It uses the Border Radius slider control.

**Comment by aristath** [comment]
> @aristath I've created a PR (#27220) to add a "rounded" block style for the search block. Does this fit with what you envisioned?

Yes, that is **exactly** what I envisioned! :heart:

**Comment by apeatling** [comment]
I think we should completely separate adding border support to Gutenberg, and how it is used on the search block. Let's merge the block support here and remove search block changes. 

From my perspective the border support is very important to allow designers to produce different kinds of patterns using different blocks ([examples](https://github.com/WordPress/gutenberg/issues/26559) [here](https://github.com/WordPress/gutenberg/issues/26556)). I would include the search block there as well, as more patterns are created for site headers for example.

**Comment by aaronrobertshaw** [comment]
I've rebased this again and removed the changes to the search block. The PR description has been updated to reflect the current state of the PR along with new testing instructions.

**Comment by aaronrobertshaw** [comment]
There is now also a new PR containing the changes to the search block for it to opt into the border radius support and adjust inner radii etc. https://github.com/aaronrobertshaw/gutenberg/pull/1

Once this PR is accepted and merged I'll update the new PR accordingly.

File: packages/block-editor/src/hooks/border.js
Comment (mcsf) [inline packages/block-editor/src/hooks/border.js:66]:
```suggestion
	return [ useIsBorderRadiusDisabled( props ) ].every( Boolean );
```

File: packages/block-editor/src/hooks/border.js
Comment (mcsf) [inline packages/block-editor/src/hooks/border.js:49]:
```suggestion
	return true === support || support?.radius;
```

If you care about the return type being `boolean`:

```suggestion
	return !! ( true === support || support?.radius );
```

File: packages/block-editor/src/hooks/border-radius.js
Comment (mcsf) [inline packages/block-editor/src/hooks/border-radius.js:67]:
```suggestion
	return !! ( true === support || support?.radius );
```

**Review by mcsf — APPROVED** [review]
@aaronrobertshaw: nice work, thanks for moving the Search Block work out of this PR and keeping it to the point. I'm tempted to :+1: this, but would like a second opinion before merging.

**Review by apeatling — APPROVED** [review]
Looks great @aaronrobertshaw, thanks for separating out the search block implementation.

**Comment by nosolosw** [comment]
I've added the "Needs Dev Note" in this issue so we don't forget to create a Dev Note for WP 5.7 for this kind of change: list all the new style properties that were added to blocks and users can update. This should be useful for themes and plugins to check whether they need to adapt anything. Hat tip to Riad for the suggestion.

File: lib/class-wp-theme-json.php
Comment (nosolosw) [inline lib/class-wp-theme-json.php:53]: The values on this array need to be the name of the property as they appear in the `PROPERTIES_METADATA`, so this should have been `borderRadius`. There's a couple of other things I'd like to share:

- these styles are attached to the `:root` element, so I'm not sure borderRadius makes a lot of sense here
- the global styles sidebar doesn't have a corresponding UI control for the user to tweak this property ― neither for the global or the group block.

In https://github.com/WordPress/gutenberg/pull/28320 I'm removing the support for `borderRadius` for the root element but could also just rename the property to fix the issue.

@aaronrobertshaw would you be available to create a follow-up PR that adds the radius control to the global styles sidebar if the block supports it?

Comment (aaronrobertshaw) [reply lib/class-wp-theme-json.php:53]: @nosolosw thank you for the explanation. I agree, a `border-radius` value on `:root` doesn't make a lot of sense. 

It was suggested via feedback on other border radius related PRs that there should be a "global" border radius value that themes could use or supply to achieve a more consistent visual result. I mention this as having the border radius as a CSS variable on the `:root` selector would help achieve this. 

In https://github.com/WordPress/gutenberg/pull/27991 I've used a similar approach adding a CSS variable for the border radius so that inner elements could adjust the value to keep uniform spacing between their border and their parent's.

It would be great to get your thoughts on that PR if you can spare the time.

> @aaronrobertshaw would you be available to create a follow-up PR that adds the radius control to the global styles sidebar if the block supports it?

I can do however I had been advised to avoid adding any further sidebar controls until https://github.com/WordPress/gutenberg/issues/27331 lands. Complicating matters and the UI, there are also requests to include support for other border properties such as width and style. I realise these aren't in place at the moment, just mentioning as they seemed to have contributed to the decision to hold off adding extra controls.

Again, I'm happy to do the follow-up if you can confirm I'm ok to do so.

Comment (nosolosw) [reply lib/class-wp-theme-json.php:53]: I've left my comments on https://github.com/WordPress/gutenberg/pull/27991 and https://github.com/WordPress/gutenberg/pull/27664#issuecomment-762909628

> I can do however I had been advised to avoid adding any further sidebar controls until #27331 lands. Complicating matters and the UI, there are also requests to include support for other border properties such as width and style. I realise these aren't in place at the moment, just mentioning as they seemed to have contributed to the decision to hold off adding extra controls.

Can you fill me up a bit on this?

Would it work if we add the UI control for the GS sidebar but we [keep](https://github.com/WordPress/gutenberg/blob/master/lib/experimental-default-theme.json#L173) the `customRadius` to false? That way we have the support built-in, themes can use it via `theme.json` but users won't see the controls. Same for the other border properties.

Comment (aaronrobertshaw) [reply lib/class-wp-theme-json.php:53]: I have a PR up adding the border radius controls to the global styles sidebar https://github.com/WordPress/gutenberg/pull/28346

> Can you fill me up a bit on this?

Some extra background on past discussions regarding controls for border related support can be found in https://github.com/WordPress/gutenberg/issues/21540#issuecomment-747282075. 

I'm happy to change approach. Whatever we need to get the best result.

**Comment by nosolosw** [comment]
As per https://github.com/WordPress/gutenberg/issues/28539#issuecomment-769754365 I've removed the "needs dev note" label from this note. Sorry about the confusion!

**Pull request merged**
